### PR TITLE
Check for hostname before setting isMac to true

### PR DESCRIPTION
**Pull request by therealbrad**
## Proposed changes

[//]: # Check if hostname is set when determining if instance is running locally. Without this check, a local mac os instance can send Command key to a remote Window instance.

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by christian-bromann** [comment]
@therealbrad thanks for taking a stab at this. Let's make sure we add proper unit tests so we don't run into any regressions moving on. There are already [existing unit tests](https://github.com/webdriverio/webdriverio/blob/main/packages/webdriverio/tests/commands/browser/action.test.ts#L58-L74) for this case. Also we need you to have the CLA signed. Thank you!

**Review by christian-bromann — COMMENTED** [review]
LGTM 👍 

Once the CLA is signed I can go ahead and merge this.

**Review by christian-bromann — APPROVED** [review]
Thank you 👍

**Comment by christian-bromann** [comment]
@therealbrad there seems to be an failure in the test?

**Comment by therealbrad** [comment]
@christian-bromann My bad. I meant to remove that test because I didn't see a way to run unit tests and set browser.options.hostname to null. Is there a way to do that?

**Comment by christian-bromann** [comment]
I thought `browser.options.hostname = 'local'` would do the trick because it is the same value as `this.instance` if not we should track down where `this.instance` is defined and find a way to modify that value for our test purposes. You are always free to write an individual unit test for the action class itself if it makes writing the test easier.

**Comment by therealbrad** [comment]
@christian-bromann I changed it to explicitly check if running locally via hostname instead of assuming it. Seems to work but I don't know if this breaks anything else.

File: packages/webdriverio/src/utils/actions/key.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default class KeyAction extends BaseAction {
             // check capabilities first
             platformName && platformName.match(/mac(\s)*os/i) ||
             // if not set, expect we run locally
-            os.type().match(/darwin/i)
+            (this.instance.options.hostname?.match(/local/i) && os.type().match(/darwin/i))
```
Comment (christian-bromann) [inline packages/webdriverio/src/utils/actions/key.ts:23]: `hostname` could also be `127.0.0.1`. I think it is impossible to really know whether the session runs on ones machine or not. So let's just check for `localhost` and `127.0.0.1`

**Comment by christian-bromann** [comment]
Thank you @therealbrad

**Pull request merged**